### DISCUSSION TITLE
Heroku: Webpack compile process checks other files outside entry list and shows error for those files

### DISCUSSION THREAD

**Discussion by ngohuytrieu** [webpack/webpack#18443] [Q&A]
# Bug report

**What is the current behavior?**
Hi guys, I am encountering this error, it is not happening on my local setup but only when I deploy my web app to `heroku`

Here is my folder structure
```
src
-- assets
---- scripts
------ admin.ts
test
-- playwright
---- utils
------ image-compare.ts
webpack-js.config.js
```

Here is my compile script: 
`webpack --config webpack-js.config.js --mode=production`

Here is my `webpack-config` file:
```js
const webpack = require('webpack');
const path = require('path');

module.exports = (env, argv) => [
  {
    entry: {
      admin: './src/assets/scripts/admin.script.ts',
    },
    output: {
      filename: '[name].bundle.js',
      path: path.resolve(__dirname, './public/dist/js'),
      clean: true,
    },
    module: {
      rules: [
        {
          test: /\.script.ts?$/,
          use: 'ts-loader',
          exclude: /node_modules|src\/test/,
        },
      ],
    },
    resolve: {
      extensions: ['.ts', '.js'],
    },
    // seperate vendor and import to seperate files
    optimization: {
      splitChunks: {
        cacheGroups: {
          vendor: {
            test: /[\\/]node_modules[\\/]/,
            name: 'vendor',
            chunks: 'all',
          },
        },
      },
    },
  },
];
```

The problem is when I run the above script, it not only compiles `admin.script.ts` file but only checking on other `.ts` files outside `src/assets/scripts` and shows error for those files, like this:
```bash
ERROR in /app/test/playwright/utils/image-compare.ts(1,24)
TS2307: Cannot find module '@playwright/test' or its corresponding type declarations.
```

I did try to add exclude `test` folder to  `exclude: /node_modules|src\/test/` but still not luck
I also set file prefix is `.script.ts` (so it would not take all `.ts` files) to `test: /\.script.ts?$/,` and still not working

**If the current behavior is a bug, please provide the steps to reproduce.**
1. Config `webpack-js.config.js`
2. Add script to package.json
3. Run that sript

**What is the expected behavior?**
Webpack only compiles files in entry list

**Other relevant information:**
webpack version:  `^5.88.2`
webpack-cli version:  `^5.1.4`
Node.js version: `^18.17.0`
Operating System: `Heroku`
Additional tools:

**Comment by alexander-akait** [comment]
This is typescript error, not webpack, you need to improve your `tsconfig.json` file

**Comment by ngohuytrieu** [comment]
@alexander-akait  It only shows those errors when I run webpack compile command
Here is my `tsconfig.json` file
```
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "target": "ES2021",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "rootDir": "./",
    "incremental": true,
    "skipLibCheck": true,
    "strictNullChecks": false,
    "noImplicitAny": false,
    "strictBindCallApply": false,
    "forceConsistentCasingInFileNames": false,
    "noFallthroughCasesInSwitch": false
  }
}

```

**Reply by alexander-akait** [reply]
Can you run `npm ls @playwright/test`, also can you try to compile it using pure typescript, without webpack?
